### PR TITLE
Custom attributes

### LINKED ISSUE #1587: Custom attributes

### PR DESCRIPTION
**Pull request by BykhovDenis**
close #1587 
Signed-off-by: Denis Bykhov <80476319+BykhovDenis@users.noreply.github.com>

### CODE REVIEW

File: packages/model/src/dsl.ts [hunk 1]
```diff
@@ -335,79 +335,99 @@ export class Builder {
...
+  _class: Ref<Class<T>>,
+  label: IntlString,
+  attrs: Record<string, any> = {}
+): T {
+  // eslint-disable-next-line
+  return {
+    space: core.space.Model,
+    modifiedBy: core.account.System,
+    modifiedOn: Date.now(),
+    _id: generateId(),
```
Comment (haiodo) [inline packages/model/src/dsl.ts]: Why we need id?

File: plugins/activity-resources/src/components/utils.ts [hunk 2]
```diff
@@ -37,14 +37,21 @@ async function createPseudoViewlet (
   // Check if it is attached doc and collection have title override.
   const presenter = await getObjectPresenter(client, doc._class, { key: 'doc-presenter' })
   if (presenter !== undefined) {
+    let collection = ''
+    if (dtx.collectionAttribute?.label !== undefined) {
+      if (dtx.collectionAttribute.isCustom) {
```
Comment (haiodo) [inline plugins/activity-resources/src/components/utils.ts]: If we have embedded why we need this logic? And it looks wrong

**Comment by haiodo** [comment]
Why we need isCustom?

File: plugins/view-resources/src/components/CreateAttribute.svelte [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  export let _class: Ref<Class<Space>>
+  let name: string
+  let type: Type<PropertyType> | undefined
+  let is: AnyComponent | undefined
+  const client = getClient()
+  const hierarchy = client.getHierarchy()
+  const dispatch = createEventDispatcher()
+
+  async function save (): Promise<void> {
+    if (type === undefined) return
```
Comment (haiodo) [inline plugins/view-resources/src/components/CreateAttribute.svelte:33]: Please show confirmation dialog and inform about changes will not be possible to perform for type and field name in document.

File: plugins/activity-resources/src/components/utils.ts (hunk 2 above)
Comment (BykhovDenis) [reply plugins/activity-resources/src/components/utils.ts]: It's outdated, Thanks

**Comment by BykhovDenis** [comment]
`Why we need isCustom?` I left it for possible change label in future

**Review by haiodo — APPROVED** [review]

**Pull request merged**